### DISCUSSION TITLE
Consider improving Jest detection behavior to avoid errors and remove need for `jest.disabledWorkspaceFolders`

### DISCUSSION THREAD

**Discussion by segevfiner** [orta/vscode-jest#886] [Ideas]
Followup discussion of https://github.com/jest-community/vscode-jest/issues/877.

If we could improve the detection logic of Jest in workspace folders, we could remove the need for `jest.disabledWorkspaceFolders` and have things just work, this will require implementing detection/lookup logic similar to Jest itself, or otherwise handling the error from it not finding configuration/tests.

**Comment by connectdotz** [comment]
I am in favor of adding a quick `exclude the workspace` button in the error panel when the jest run failed and it is a multi-root workspace. That way we don't need to duplicate the jest detection logic or implement a complex-time-consuming-yet-incomplete detection logic.

what do you guys think?

**Reply by segevfiner** [reply]
But in which config file you want it to add the setting? If we can somehow just catch the detection error from Jest or reuse their logic if they are using some library...

**Reply by segevfiner** [reply]
I wonder how the vitest extension handles this
